### PR TITLE
[ADD] module pricelist_per_product

### PR DESCRIPTION
**Pull request by bealdav**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
One alernative we have developed for this is: https://github.com/odoomrp/odoomrp-wip/tree/8.0/product_supplierinfo_for_customer

You can check to see if you like it

**Comment by bealdav** [comment]
Thanks to point it, nice feature too. OCA is wonderful

**Comment by bealdav** [comment]
I have just to improve translation

**Comment by pedrobaeza** [comment]
Please don't put company logo.

File: pricelist_per_product/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Pricelist Per Product',
+    'version': '0.1',
```
Comment (pedrobaeza) [inline pricelist_per_product/__openerp__.py]: Better 1.0

**Comment by bealdav** [comment]
Why ?

**Comment by bealdav** [comment]
1/ Marketing of **free** modules begins to be more important now
https://www.odoo.com/apps/modules?price=Paid
Paid modules will be promoted in next days even if poor quality 
see https://twitter.com/bealdav/status/568704010422751232, 
OCA must promote its contibutors

2/ Akretion is OK to maintain this module alone if required.

3/ Akretion promote OCA ecosystem and OCA partners.
http://www.akretion.com/fr/blog/odoo-erp-module-connection-a-un-centre-logistique

see sentence "C'est d'ailleurs un gage de pérennité pour vos systèmes d'informations que d'utiliser ces modules, développés et validés par ces experts contributeurs de l'OCA"
that can be translate "This is also a guarantee of continuity for your information systems to use these modules, developed and validated by the expert contributors OCA"

**Comment by pedrobaeza** [comment]
Because it's an OCA rule. Your name will be in author and contributors section, but company logos are not allowed in the module icon, because this icon is for diplaying visually the function of the module or a visual identification for it.

**Comment by pedrobaeza** [comment]
Also, if you have 20 modules with the same icon (your company logo), the visual differentation between them will be nothing. Better no icon than the same one.

**Comment by bealdav** [comment]
OK, can I put the OCA logo in the left of the icon instead of a§

**Comment by pedrobaeza** [comment]
I don't understand your proposal. What do you want to do?

**Comment by bealdav** [comment]
icon copie

ok for that ?

**Comment by pedrobaeza** [comment]
No, I think it's preferable not to have anything, than this possible duplicated logo across modules. We have the OCA logo in the description if you use OCA template (https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst).

See some examples for proper module icons:
- https://github.com/OCA/web/blob/8.0/web_sheet_full_width/static/description/icon.png
- https://github.com/OCA/server-tools/blob/8.0/disable_openerp_online/static/description/icon.png
- https://github.com/OCA/server-tools/blob/8.0/auth_admin_passkey/static/description/icon.png

Or some I have made myself for slightly differentiate a set of modules about the same topic:
- https://github.com/odoomrp/odoomrp-wip/blob/8.0/quality_control/static/description/icon.png
- https://github.com/odoomrp/odoomrp-wip/blob/8.0/quality_control_mrp/static/description/icon.png
- https://github.com/odoomrp/odoomrp-wip/blob/8.0/quality_control_between_dates/static/description/icon.png
- https://github.com/odoomrp/odoomrp-wip/blob/8.0/quality_control_claim/static/description/icon.png

Do you see the visual goal?

**Comment by bealdav** [comment]
you're right, done

**Comment by pedrobaeza** [comment]
:+1:

**Comment by chafique-delli** [comment]
It's ok for me but in the evolution of the module, it would be that the module also provides the pricelists on existing products. :+1:

**Comment by florian-dacosta** [comment]
:+1:

**Comment by LeartS** [comment]
I installed the module, Checked "Product Price Grid" on my pricelists, but in the product form view I see the grid with no rows.

**Comment by bealdav** [comment]
@LeartS create a new product, the grid will be appear

**Comment by LeartS** [comment]
@bealdav And I can't see the grid for already existing products?

Maybe I did not understand what this module is supposed to do, I thought it shows the price of a product calculated using the various pricelists. Am I correct?

**Comment by bealdav** [comment]
@LeartS, you are right, there is no good solution to populate prices on existing products.
I will code this week a solution on pricelist version with filtering on product fields for select products, but I think this PR can be as it for now.

File: pricelist_per_product/static/description/index.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+<section class="oe_container">
+    <div class="oe_row oe_spaced">
+        <h2 class="oe_slogan">Pricelist Grid</h2>
+        <h3 class="oe_slogan">All pricelists at once</h3>
+        <div class="oe_span6">
+            <img class="oe_picture oe_screenshot" src="product.png">
+        </div>
+        <div class="oe_span6">
+            <p class="oe_mt32">
+Fulfill all your pricelists directly in your product form.
```
Comment (gurneyalex) [inline pricelist_per_product/static/description/index.html]: s/Fullfill/Fill in/

to fullfill c'est "remplir" au sens de "remplir une obligation", "satisfaire une condition"

File: pricelist_per_product/README.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
+Pricelist Per Product
+======================
+
+This module allows to see all sale prices on product form
```
Comment (gurneyalex) [inline pricelist_per_product/README.rst]: suggestion: s/to see/to see and edit/

s/on product form/on the product form./

File: pricelist_per_product/README.rst [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+Installation
+============
+
+One click installation
```
Comment (gurneyalex) [inline pricelist_per_product/README.rst]: I'd remove this paragraph

File: pricelist_per_product/README.rst [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+ * Go to the product form (Sales tab)
+
+.. image:: static/description/product.png
+
+
+
+Known issues / Roadmap
+======================
+
```
Comment (gurneyalex) [inline pricelist_per_product/README.rst]: maybe mention here the current issue with existing product?

File: pricelist_per_product/pricelist.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+from openerp import models, fields
+
+
+class ProductPricelist(models.Model):
+    _inherit = 'product.pricelist'
+
+    price_grid = fields.Boolean(
+        string='Product Price Grid',
+        help="Define if the price list items are fulfilled from product form\n"
```
Comment (gurneyalex) [inline pricelist_per_product/pricelist.py]: s/fullfilled/filled in/ (see other comment about the verb to fullfill)

File: pricelist_per_product/README.rst (hunk 4 above)
Comment (bealdav) [inline pricelist_per_product/README.rst]: @pedrobaeza @gurneyalex  Isn't a mandatory paragraph ?
Maybe template should be convert: 'Installation' -> 'Specific Installation Instructions'

File: pricelist_per_product/README.rst (hunk 5 above)
Comment (bealdav) [inline pricelist_per_product/README.rst]: I'm being resolve it

File: pricelist_per_product/static/description/index.html (hunk 2 above)
Comment (bealdav) [inline pricelist_per_product/static/description/index.html]: done, thanks

File: pricelist_per_product/README.rst (hunk 4 above)
Comment (gurneyalex) [inline pricelist_per_product/README.rst]: IMO, nothing is "mandatory". If you don't have anything useful to put in there, just don't put the section. 

I hate copy pasted docstrings which give no useful information but are there because of a policy saying "you have to put a docstring"...

**Comment by gurneyalex** [comment]
I added ty e WIP label since the last commit is WIP.

**Comment by legalsylvain** [comment]
@bealdav : what is the state of this PR ? Is it still WIP ?
I could be interested by this feature and I can help if necessary.

Kind regards.

File: pricelist_per_product/README.rst [hunk 7]
```diff
@@ -0,0 +1,53 @@
+Pricelist Per Product
+======================
```
Comment (max3903) [inline pricelist_per_product/README.rst]:
```
.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
   :alt: License: AGPL-3

=====================
Pricelist Per Product
=====================
```

File: pricelist_per_product/README.rst [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+Usage
+=====
+
+* Go to the product form (Sales tab): add a Price version
+
+.. image:: static/description/product.png
+
+* Menu Sales / Configuration / Pricelist Versions
+
```
Comment (max3903) [inline pricelist_per_product/README.rst]:
```
.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
   :alt: Try me on Runbot
   :target: https://runbot.odoo-community.org/runbot/135/8.0
```

**Comment by max3903** [comment]
Tested on runbot. Once minor issues are fixed :+1:

File: product_gtin/product_gtin.py [hunk 9]
```diff
@@ -19,11 +19,11 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-import logging
-_logger = logging.getLogger(__name__)
 
 from openerp.osv import orm, fields
 import operator
+import logging
+_logger = logging.getLogger(__name__)
```
Comment (gurneyalex) [inline product_gtin/product_gtin.py]: unrelated change.

File: pricelist_per_product/pricelist.py [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    currency_name = fields.Many2one(
+        related='price_version_id.pricelist_id.currency_id',
+        readonly=True)
+    related_sequence = fields.Integer(
+        String='Sequence',
+        related="sequence")
+
+    @api.multi
+    def button_product(self):
+        if self.product_tmpl_id:
```
Comment (gurneyalex) [inline pricelist_per_product/pricelist.py:80]: missing loop on `self` or self.ensure_one()

**Comment by gurneyalex** [comment]
:+1:

**Comment by sebastienbeau** [comment]
Thanks David for the work !
